### PR TITLE
Feature/stm32cube

### PR DESCRIPTION
**Pull request by hefloryd**
Add support for STM32F769I-DISCO board, running STM32Cube firmware including FreeRTOS and lwIP. SNMP is currently not supported.

This PR also splits both rt-kernel and stm32cube sample apps so that board-specific support is in a separate file.

Finally, this PR changes the tutorial to enable flashing of LED1 by default, so that one can easily see if the tutorial is working.

### CODE REVIEW

File: src/ports/STM32Cube/pnal.c [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+int pnal_set_ip_suite (
+   const char * interface_name,
+   const pnal_ipaddr_t * p_ipaddr,
+   const pnal_ipaddr_t * p_netmask,
+   const pnal_ipaddr_t * p_gw,
+   const char * hostname,
+   bool permanent)
+{
+   ip_addr_t ip_addr;
```
Comment (pyhys) [inline src/ports/STM32Cube/pnal.c:41]: No checking of correct interface_name?

File: src/ports/STM32Cube/sampleapp_main.c [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+static app_data_t * sample_app = NULL;
+static pnet_cfg_t pnet_cfg = {0};
+app_args_t app_args = {0};
+
+/****************************** Main ******************************************/
+
+int _main (void)
+{
+   int ret;
+   int32_t app_log_level = APP_LOG_LEVEL;
```
Comment (pyhys) [inline src/ports/STM32Cube/sampleapp_main.c:47]: Why is this needed? Do we plan to change log level during runtime?

File: src/ports/STM32Cube/sampleapp_main.c [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+   APP_LOG_INFO ("Network interfaces:   %s\n", app_args.eth_interfaces);
+   APP_LOG_INFO ("Default station name: %s\n", app_args.station_name);
+
+   app_pnet_cfg_init_default (&pnet_cfg);
+
+   /* Note: station name is defined by app_gsdml.h */
+
+   strcpy (pnet_cfg.file_directory, APP_DEFAULT_FILE_DIRECTORY);
+
+   /* Note: pnal_cfg not is used for rt-kernel  */
```
Comment (pyhys) [inline src/ports/STM32Cube/sampleapp_main.c:73]: Update or remove comment?

**Review by pyhys — APPROVED** [review]
Great!
A few minor comments.

File: src/ports/STM32Cube/sampleapp_main.c (hunk 2 above)
Comment (hefloryd) [reply src/ports/STM32Cube/sampleapp_main.c:47]: This was copied from rt-kernel/sampleapp_main.c. Will remove in both files.

File: src/ports/STM32Cube/sampleapp_main.c (hunk 3 above)
Comment (hefloryd) [reply src/ports/STM32Cube/sampleapp_main.c:73]: This was copied from rt-kernel/sampleapp_main.c. Will remove in both files.

File: src/ports/STM32Cube/pnal.c (hunk 1 above)
Comment (hefloryd) [reply src/ports/STM32Cube/pnal.c:41]: This was copied from rt-kernel/pnal.c. Will leave unchanged for now.

File: src/ports/STM32Cube/pnal.c [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#include "pnal.h"
+
+#include "osal.h"
+#include "osal_log.h"
+#include "options.h"
```
[the remark this answers is not in the document]
Comment (hefloryd) [reply src/ports/STM32Cube/pnal.c:20]: This was copied from rt-kernel/pnal.c. Will change order in all pnal.c.

File: doc/getting_started_freertos.rst [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+repository::
+
+    CPU=cortex-m7fd BOARD=STM32F769I-DISCO CUBE_DIR=/path/to/cube cmake \
+        -B build.cube \
+        -DCMAKE_TOOLCHAIN_FILE=cmake/tools/toolchain/stm32cube.cmake \
+        -G "Unix Makefiles"
+    cmake --build build.cube
+
+When the build completes you can find the sample-app binary in
+``build.cube/pn_dev.bin``. Flash this binary to your board by
```
[the remark this answers is not in the document]
Comment (hefloryd) [reply doc/getting_started_freertos.rst:35]: Yes, the file is pn_dev.bin in the build.cube folder.
